src/transforms/detect_exceptions/mod.rs

```diff
@@ -0,0 +1,382 @@
+}
+
+impl Default for DetectExceptionsConfig {
+    fn default() -> Self {
+        Self {
+            languages: vec![ProgrammingLanguages::All],
+            expire_after_ms: default_expire_after_ms(),
+            flush_period_ms: default_flush_period_ms(),
+            multiline_flush_interval_ms: default_multiline_flush_interval_ms(),
+            max_bytes: default_max_bytes_size(),
+            max_lines: default_max_lines_num(),
+            group_by: vec![],
+        }
+    }
+}
+
+fn default_programming_languages() -> Vec<ProgrammingLanguages> {
+    Vec::from([ProgrammingLanguages::All])
+}
+
+const fn default_expire_after_ms() -> Duration {
+    Duration::from_millis(30000)
+}
+
+const fn default_flush_period_ms() -> Duration {
+    Duration::from_millis(1000)
+}
+
+const fn default_multiline_flush_interval_ms() -> Duration {
+    Duration::from_millis(1000)
+}
+
+const fn default_max_bytes_size() -> usize {
+    0
+}
+
+const fn default_max_lines_num() -> usize {
+    1000
+}
+
+impl_generate_config_from_default!(DetectExceptionsConfig);
+inventory::submit! {
+    TransformDescription::new::<DetectExceptionsConfig>("detect_exceptions")
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "detect_exceptions")]
+impl TransformConfig for DetectExceptionsConfig {
+    async fn build(&self, _context: &TransformContext) -> crate::Result<Transform> {
+        DetectExceptions::new(self).map(Transform::event_task)
+    }
+
+    fn input(&self) -> Input {
+        Input::log()
+    }
+
+    fn outputs(&self, _: &schema::Definition) -> Vec<Output> {
+        vec![Output::default(DataType::Log)]
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "detect_exceptions"
+    }
+}
+
+pub struct DetectExceptions {
+    accumulators: HashMap<Discriminant, TraceAccumulator>,
+    languages: Vec<ProgrammingLanguages>,
+    expire_after: Duration,
+    flush_period: Duration,
+    multiline_flush_interval: Duration,
+    max_bytes: usize,
+    max_lines: usize,
+    group_by: Vec<String>,
+}
+
+impl DetectExceptions {
+    pub fn new(config: &DetectExceptionsConfig) -> crate::Result<Self> {
+        if config.languages.is_empty() {
+            return Err("languages cannot be empty".into());
+        }
+        Ok(DetectExceptions {
+            accumulators: HashMap::new(),
+            languages: config.languages.clone(),
+            group_by: config.group_by.clone().into_iter().collect(),
+            expire_after: config.expire_after_ms,
+            multiline_flush_interval: config.multiline_flush_interval_ms,
+            max_bytes: config.max_bytes,
+            max_lines: config.max_lines,
+            flush_period: config.flush_period_ms,
+        })
+    }
+
+    fn consume_one(&mut self, output: &mut Vec<Event>, e: Event) {
+        let log_event = e.into_log();
+        let discriminant = Discriminant::from_log_event(&log_event, &self.group_by);
+
+        if !self.accumulators.contains_key(&discriminant) {
```

Comment:
If instead of
```
if !.contains_key() {
    .insert()
}
```
we do `.try_insert()`, we save a map lookup on the hot path.

Reply:
Plus, if `try_insert()` fails, it returns the existing (K,V). That would've worked well if TraceAccumulator::new() were cheap which it isn't. 
This still saves a lookup:
```
let accumulator = if let Some(accumulator) =  .get_mut() {
   accumulator
} else {
   let accumulator = TraceAccumulator::new(...)
   .insert()
   accumulator
};
accumulator.push(...)
```

Reply:
using `try_insert` results in error
```
error[E0658]: use of unstable library feature 'map_try_insert'
```

Reply:
That's a bummer, but try_insert() asks for construction of a TraceAccumulator every time through, which is expensive to do if we don't end up using it. Do you think the second option would work?

Reply:
i tried replacing with
```
        let accumulator = self
            .accumulators
            .entry(discriminant)
            .or_insert(TraceAccumulator::new(
                self.languages.clone(),
                self.multiline_flush_interval,
                self.max_bytes,
                self.max_lines,
            ));
```
but this didnt work as expected. So i reverted the changes back. If really needed to simplify the expression, we can do it in a separate PR.